src/chrome/internal/locations/location.ts

```diff
@@ -0,0 +1,157 @@
+import * as Validation from '../../../validation';
+import { IScript } from '../scripts/script';
+import { ISource } from '../sources/source';
+import { ILoadedSource } from '../sources/loadedSource';
+import { logger } from 'vscode-debugadapter';
+import { ColumnNumber, LineNumber, URLRegexp } from './subtypes';
+import { CDTPScriptUrl } from '../sources/resourceIdentifierSubtypes';
+import { IResourceIdentifier, parseResourceIdentifier, URL } from '../sources/resourceIdentifier';
+
+export type integer = number;
+
+export class Coordinates {
+    public isSameAs(location: Coordinates): boolean {
+        return this.lineNumber === location.lineNumber
+            && this.columnNumber === location.columnNumber;
+    }
+
+    public toString(): string {
+        return this.columnNumber !== undefined
+            ? `${this.lineNumber}:${this.columnNumber}`
+            : `${this.lineNumber}`;
+    }
+
+    constructor(
```

Comment:
Minor point of style, I prefer putting the constructor at the top. Especially helpful so I can see what the members are first before looking at methods.

Reply:
@roblourens Do you want to add a lint rule that validates that the constructors follow that convention?

Reply:
Sure but I don't feel that strongly about it.

Reply:
I'll try my best to remember it then...